src/plugins/data/common/search/aggs/buckets/histogram.ts

```diff
@@ -152,6 +156,29 @@ export const getHistogramBucketAgg = ({
             esTypes: aggConfig.params.field?.spec?.esTypes || [],
           });
         },
+        serialize(val, aggConfig) {
+          // store actually used auto interval in serialized agg config to be able to read it from the result data table meta information
+          const autoBounds = aggConfig?.getAutoBounds();
+          if (val === autoInterval && aggConfig && autoBounds) {
+            const usedInterval = calculateHistogramInterval({
+              values: autoBounds,
+              interval: aggConfig.params.interval,
+              maxBucketsUiSettings: getConfig(UI_SETTINGS.HISTOGRAM_MAX_BARS),
+              maxBucketsUserInput: aggConfig.params.maxBars,
+              intervalBase: aggConfig.params.intervalBase,
+              esTypes: aggConfig.params.field?.spec?.esTypes || [],
+            });
+            return buildSerializedAutoInterval(usedInterval);
```

Comment:
we could create a new property on this aggConfig `calculatedInterval` and make its serialize method:
```
      serialize: (val, aggConfig) => {
          const autoBounds = aggConfig?.getAutoBounds();
          if (aggConfig && autoBounds && aggConfig.params.interval === autoInterval ) {
            const usedInterval = calculateHistogramInterval({
              values: autoBounds,
              interval: aggConfig.params.interval,
              maxBucketsUiSettings: getConfig(UI_SETTINGS.HISTOGRAM_MAX_BARS),
              maxBucketsUserInput: aggConfig.params.maxBars,
              intervalBase: aggConfig.params.intervalBase,
              esTypes: aggConfig.params.field?.spec?.esTypes || [],
            });
            return usedInterval;
         }
         return aggConfig.params.interval;
       }
```

we don't need any deserialize method for it, write should be _.noop